frontend/justfile

```diff
@@ -71,3 +71,19 @@ run *args:
 
 types:
     cd .. && pnpm exec vue-tsc -p frontend --noEmit
+
+##########
+# Health #
+##########
+
+# Check the health of the service
+@health host="localhost:50288":
+    -curl -s -o /dev/null -w '%{http_code}' 'http://{{ host }}/api/health'
+
+# Wait for the service to be healthy
+@wait host="localhost:50288":
+    # The just command on the second line is executed in the context of the
+    # parent directory and so must be prefixed with `api/`.
+    just ../_loop \
+    '"$(just frontend/health {{ host }})" != "200"' \
```

Comment:
I think we should also check that the response contains 3 `"ok"`s. 
```json
{
  "clickhouse": "ok",
  "postgres": "ok",
  "sites_cache": "ok"
}
```

I haven't seen what the status code and response is when Plausible is not healthy so there is a possibility that it returns some "not ok" but with a HTTP 200 response.

Reply:
@dhruvkb I found a two ways the parse the JSON object. First option, we can use a `jq` which is a JSON parsing and transformation tool for linux and determine if the response contains 3 `"ok"`s. Second options, we can create python to parse through the JSON object and determine  if the response contains 3 `"ok"`s. Let me know if there is better solution. What is the best option to take?

Reply:
I don't want to add a dependency on `jq`. Using Python might be fine 👍.

Reply:
Agreed that not adding jq as a dependency is good. We've avoided it in other instances as well, opting instead for Python then too.

Reply:
I think we don't need something as complex as Python or jq or even JSON parsing. Looking at the [plausible code for the healthcheck endpoint](https://github.com/plausible/analytics/blob/master/lib/plausible_web/controllers/api/external_controller.ex#L50-L80), we can just as easily do this and it would catch the unhealthy cases:

```
curl -s 'http://localhost:50288/api/health' --fail | grep -vq 'error'
```

`-vq` inverts the match and suppresses grep output (stops it from printing matched lines, in this case all the `ok` that don't match `error`, inverted) so we just get the exit code. `--fail` in curl causes it to fail if the response code is 500, bypassing grep entirely.

Reply:
@sarayourfriend your solves only require curl and grep which I would prefer.

Reply:
@dhruvkb Please take a look at the suggested resolution to use grep and invert matching.

Reply:
Sorry for not responding here. You'd need to use the exit code for my original suggestion, which you can get with `$?`. Your approach looks more conventional though, I always have to look up what `$?` is :sweat_smile:

Reply:
I tried `'http://localhost:50288/api/health' --fail | grep -vq 'error' | echo $?` and getting this error:
 ```
--rpc-eval : RPC failed with reason :nodedown
error: Recipe `init` failed on line 66 with exit code 1
```
Also, I'm not sure how I can pass a exit code into a command substitution.

It seems like the `curl -s 'http://localhost:50288/api/health' --fail | grep -v -c 'error'` is working as expected. Let me know if there could be any edge case. 

What are the other API status result from `clickhouse`, `postgres` and `sites_cache` beside "ok" and "error"?